src/services/publisher_service/helpers.rs

```diff
@@ -173,3 +173,23 @@ pub async fn update_message_processing_status(
         .await?;
     Ok(())
 }
+
+/// Checks for messages in the `processing` state for more than threshold in minutes
+/// and put it back in a `queued` state for processing
+#[instrument(skip(postgres))]
+pub async fn dead_letters_check(
```

Comment:
Can we get some basic storage tests for this before merging?

Reply:
I'll add tests for `dead_letters_check ` and `dead_letter_give_up_check` asap, but requesting for a review now to be unblocked with the implementation itself. 
I've added a subtask for this in the PR description to not miss this before merging.